coins/monero/src/block.rs

```diff
@@ -65,6 +68,29 @@ impl Block {
     Ok(())
   }
 
+  pub fn tx_merkle_root(&self) -> [u8; 32] {
+    merkle_root::tree_hash(self.miner_tx.hash(), &self.txs)
+  }
+
+  pub fn serialize_hashable(&self) -> Vec<u8> {
+    let mut blob = self.header.serialize();
+
+    blob.extend_from_slice(&self.tx_merkle_root());
+
+    write_varint(&(1 + self.txs.len() as u64), &mut blob).unwrap();
```

Comment:
yep will change that but wouldn't using `as` here be safe as long as there aren't any 128 bit CPUs

Reply:
There are quite a few execution units exceeding 64-bit widths. Rust compatibility for them, and memory which would cause a pointer to exceed 2**64, is a distinct countermeasure to this being an issue.